### ISSUE TITLE
some problems with reproducing the results

### ISSUE THREAD

**Issue by DavidQiuChao** [rqhuang88/DV-Matcher#2]
P1:
When training with the default settings (just running the command: "python train.py"), the training loss and validating loss are both increasing after several epochs. 

![Image](https://github.com/user-attachments/assets/4dfc023e-6192-4979-9503-790cd776f6c2) ![Image](https://github.com/user-attachments/assets/84e7ddd7-9a3d-438e-b542-b8e5430fd08e)

![Image](https://github.com/user-attachments/assets/dd4484c9-37b1-46f2-950d-360b0138efd2)

P2:
When evaluating the results, where can I get the paths in "main.m" file ?

![Image](https://github.com/user-attachments/assets/19dede54-5185-4c4c-96e9-2780609d29be)

**Comment by zhangquanchen** [comment]
1. Maybe you have something wrong (like some code change? or data problem?) Or you can simply change the hyperpara and then have more trial? **I think if you still have some problems, we can arrange a simple zoom meeting to debug**. Our tensorboard result is actually as followed:

![Image](https://github.com/user-attachments/assets/3fb72f53-d628-40bc-9e33-b79f80264a0c)

![Image](https://github.com/user-attachments/assets/16298b6b-1a6e-4ae1-92c6-ed4624c427c0)

![Image](https://github.com/user-attachments/assets/982f31db-f0f3-4d8c-ba73-6f5f43b196b9)

2. 
(1) shape: this is the shapes you can find in `data/scape_r/shapes_test`, others are in `https://drive.google.com/drive/folders/1CK9qihI2yyxkuXsxSHqzTRhLdTk8qghn`
(2) vts: this is the gt correspondence for eval (you can also find in original dataset). I have added it under `data/scape_r/corres`.
(3) M: this is the geodist matrix, you can get by running `python eval/geo_mat.py` on the specific shape.
(1) phiS, phiT: After run `python test_full.py` with specific ckpt, you will find the **feature result** under folder `result/xxx`, where xxx is your exp name.

**Comment by DavidQiuChao** [comment]
Hi，I re-downloaded the project and re-trained the model and got the loss curve as follows:

![Image](https://github.com/user-attachments/assets/11ae9c51-f7dd-447e-89ed-5a2137482aa3) ![Image](https://github.com/user-attachments/assets/b800d364-83b6-46b1-bffe-62a9b82ec743)

![Image](https://github.com/user-attachments/assets/8f35bd4d-b6f1-4e37-8bf1-a1dcfd7cb7c5)

The loss was increasing around 16th epoch。I was not sure it matters. So I tried to use the evaluation code to get the metrics. But, I got errors in as follows:

![Image](https://github.com/user-attachments/assets/5c88fc59-287f-4da3-bc18-9276951657b1)
I found the shapes were not match. Beacuse the shapes below were not the same for every mesh:

![Image](https://github.com/user-attachments/assets/16e593c7-9307-4498-966f-67cf0033d611)

II hope you can give me some helps。

**Comment by zhangquanchen** [comment]
Hi, this error is due to the different **MATLAB version**. Some MATLAB versions cannot directly assign values between matrices of different dimensions. You can try the following  compatible code instead (others are the same):
`
[num_rows, ~] = size(basis_full);
for i = 1:num_rows
    if Partial_idx_S(i) > 0
        basis_full(i,:) = phiS(i,:);
    end
end
`
If there are other problems, please let me know.

By the way, you can also check the deformation results, which is more intuitive.

**Comment by DavidQiuChao** [comment]
> Hi, this error is due to the different **MATLAB version**. Some MATLAB versions cannot directly assign values between matrices of different dimensions. You can try the following compatible code instead (others are the same):
> 
> ```
> [num_rows, ~] = size(basis_full);
> for i = 1:num_rows
>     if Partial_idx_S(i) > 0
>         basis_full(i,:) = phiS(i,:);
>     end
> end
> ```
> 
> If there are other problems, please let me know.
> 
> By the way, you can also check the deformation results, which is more intuitive.

It looks like all the indexing operations are failed, because the "src" variable contains index beyond 5000. 

![Image](https://github.com/user-attachments/assets/41354457-3b9e-4b09-b66f-b00e8ee912b4)

![Image](https://github.com/user-attachments/assets/1199bb05-dfb5-46b3-b15a-1b10ff84e053)
